site/src/component/RecentOfferings/RecentOfferings.tsx

```diff
@@ -0,0 +1,75 @@
+import { FC } from 'react';
+import { Check } from 'react-bootstrap-icons';
+import './RecentOfferings.scss';
+
+interface RecentOfferingsProps {
+  terms: string[];
+}
+
+function parseOfferings(terms: string[]): { [academicYear: string]: boolean[] } {
+  const offerings: { [academicYear: string]: boolean[] } = {};
+
+  for (const term of terms) {
+    const [yearStr, quarter] = term.split(' ');
+    const year = parseInt(yearStr, 10);
+
+    let quarterIndex = -1;
+    if (quarter.startsWith('Fall')) quarterIndex = 0;
+    else if (quarter.startsWith('Winter')) quarterIndex = 1;
+    else if (quarter.startsWith('Spring')) quarterIndex = 2;
+    else if (quarter.startsWith('Summer')) quarterIndex = 3;
+    else continue;
+
+    // if the course is not described as a "Fall" course, it should be listed as the previous academic year
+    // e.g. "Winter 2023" should be in "2022-2023", but "Fall 2023" should be in "2023-2024"
+    const academicYear = quarterIndex === 0 ? `${year}-${year + 1}` : `${year - 1}-${year}`;
```

Comment:
I prefer the term `startYear` instead of `academicYear`, and I think that's also what we use for Zot4Plan imports.